src/main/java/org/mule/extensions/jms/internal/source/JmsListener.java

```diff
@@ -186,6 +184,8 @@ public void onStart(SourceCallback<Object, JmsAttributes> sourceCallback) throws
     connection = connectionProvider.connect();
     jmsSupport = connection.getJmsSupport();
 
+    connection.registerExceptionListener(e -> sourceCallback.onConnectionException(new ConnectionException(e)));
```

Comment:
consider that e might be a ConnectionException already

Reply:
`ExceptionUtils.extractConnectionException(e).orElseGet(() -> new COnnectionException(e))`

Reply:
no, this will never throw `ConnectionException`, are just exceptions thrown by the JMS Client